Realm/RLMResults.mm

```diff
@@ -563,6 +571,134 @@ - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMResults *, RLMCollec
     return _results.add_notification_callback(RLMWrapCollectionChangeCallback(block, self, true), std::move(keyPaths));
 }
 
+- (void)completionWithThreadSafeReference:(RLMThreadSafeReference * _Nullable)reference
+                                    queue:(dispatch_queue_t _Nullable)queue
+                               completion:(RLMResultsCompletionBlock)completion
+                                    error:(NSError *_Nullable)error {
+    auto tsr = (error != nil) ? nil : reference;
+    RLMRealmConfiguration *configuration = _realm.configuration;
+    if (queue) {
+        RLMScheduler* scheduler = [RLMScheduler dispatchQueue:queue];
+        if (reference) {
+            [scheduler invoke:^{
+                RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration queue:queue error:nil];
+                RLMResults *collection = [realm resolveThreadSafeReference:tsr];
+                collection.associatedSubscriptionId = self.associatedSubscriptionId;
+                [realm refresh];
```

Comment:
This should not be calling `refresh`.

Reply:
This is the only way the realm is updated, if we complete the subscription on a different calling thread

Reply:
I do not understand this explanation.

Reply:
I though we need to get the callback confined realm to the state of the realm which is running the subscription, (which is the calling thread realm)